### PR TITLE
Make `SiteFavicon` a shared component

### PR DESCRIPTION
**Pull request by okmttdhr**
Related to 
- https://github.com/Automattic/dotcom-forge/issues/10138
- https://github.com/Automattic/wp-calypso/pull/97635, https://github.com/Automattic/wp-calypso/pull/97593, https://github.com/Automattic/wp-calypso/pull/97542

## Proposed Changes

This PR makes `SiteFavicon` a shared component, extracting from A4A. 

## Why are these changes being made?

- https://github.com/Automattic/dotcom-forge/issues/10138
- Please refer to the Project Thread pbxlJb-6J5-p2 and the proposal post pfYzsZ-Y4-p2 for more context.

## Testing Instructions

both on Calypso and A4A Live

- Go to /site 
- Click any site 
- Ensure nothing breaks 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/sites/components/sites-dataviews/utils.ts
Comment (yashwin) [inline client/sites/components/sites-dataviews/utils.ts:1]: Is this supposed to be on this PR?

Comment (okmttdhr) [reply client/sites/components/sites-dataviews/utils.ts:1]: Sorry, I'll make a separate one!

**Review by yashwin — APPROVED** [review]
LGTM!

I could see the site icon being displayed on WordPress.com/sites when I open the site details and no regression found on A4A.

I left one non-blocking comment.

File: client/sites/components/sites-dataviews/utils.ts
Comment (okmttdhr) [reply client/sites/components/sites-dataviews/utils.ts:1]: https://github.com/Automattic/wp-calypso/pull/97684

**Pull request merged**